hyperstart/api/json/vm_spec.go

```diff
@@ -0,0 +1,102 @@
+package json
+
+type VmVolumeDescriptor struct {
+	Device       string `json:"device"`
+	Addr         string `json:"addr,omitempty"`
+	Mount        string `json:"mount"`
+	Fstype       string `json:"fstype,omitempty"`
+	ReadOnly     bool   `json:"readOnly"`
+	DockerVolume bool   `json:"dockerVolume"`
+}
+
+type VmFsmapDescriptor struct {
+	Source       string `json:"source"`
+	Path         string `json:"path"`
+	ReadOnly     bool   `json:"readOnly"`
+	DockerVolume bool   `json:"dockerVolume"`
+}
+
+type VmEnvironmentVar struct {
+	Env   string `json:"env"`
+	Value string `json:"value"`
+}
+
+type VmProcess struct {
+	// Terminal creates an interactive terminal for the process.
+	Terminal bool `json:"terminal"`
+	// Sequeue number for stdin and stdout
+	Stdio uint64 `json:"stdio,omitempty"`
+	// sequeue number for stderr if it is not shared with stdout
+	Stderr uint64 `json:"stderr,omitempty"`
+	// Args specifies the binary and arguments for the application to execute.
+	Args []string `json:"args"`
+	// Envs populates the process environment for the process.
+	Envs []VmEnvironmentVar `json:"envs,omitempty"`
+	// Workdir is the current working directory for the process and must be
+	// relative to the container's root.
+	Workdir string `json:"workdir"`
+}
+
+type VmContainer struct {
+	Id            string               `json:"id"`
+	Rootfs        string               `json:"rootfs"`
+	Fstype        string               `json:"fstype,omitempty"`
+	Image         string               `json:"image"`
+	Addr          string               `json:"addr,omitempty"`
+	Volumes       []VmVolumeDescriptor `json:"volumes,omitempty"`
+	Fsmap         []VmFsmapDescriptor  `json:"fsmap,omitempty"`
+	Sysctl        map[string]string    `json:"sysctl,omitempty"`
+	Process       VmProcess            `json:"process"`
+	Entrypoint    []string             `json:"-"`
```

Comment:
remove Entrypoint directly.
it is referenced in one place only. it is assigned without any other reference. so it can be removed.
correct me if I was wrong.